File: backend-python/app/modules/system/entity/sys_role_menu.py
from sqlalchemy import BigInteger
from sqlalchemy.orm import Mapped, mapped_column

from app.modules.system.entity.base import Base


class SysRoleMenu(Base):
    __tablename__ = "sys_role_menu"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    role_id: Mapped[int] = mapped_column(BigInteger)
    menu_id: Mapped[int] = mapped_column(BigInteger)

File: backend-go/internal/modules/system/entity/sys_role_menu.go
package entity

import (
	"drip-admin/backend-go/internal/common"
	"time"
)

type SysRoleMenu struct {
	ID        common.Int64String `gorm:"column:id;primaryKey" json:"id"`
	RoleID    common.Int64String `gorm:"column:role_id" json:"roleId"`
	MenuID    common.Int64String `gorm:"column:menu_id" json:"menuId"`
	CreatedAt *time.Time         `gorm:"column:created_at" json:"createdAt"`
}

func (SysRoleMenu) TableName() string { return "sys_role_menu" }
